packages/apollo-server-lambda/src/ApolloServer.ts

```diff
@@ -159,6 +160,35 @@ export class ApolloServer extends ApolloServerBase {
         });
       }
 
+      if (event.path === '/.well-known/apollo/server-health') {
+        const successfulResponse = {
+          body: JSON.stringify({ status: 'pass' }),
+          statusCode: 200,
+          headers: {
+            'Content-Type': 'application/json',
+            ...requestCorsHeadersObject,
+          },
+        };
+        if (onHealthCheck) {
+          onHealthCheck(event)
+            .then(() => {
+              return callback(null, successfulResponse);
+            })
+            .catch(() => {
+              return callback(null, {
+                body: JSON.stringify({ status: 'fail' }),
+                statusCode: 200,
```

Comment:
I think this status code should be 503 — do you agree?

```suggestion
                statusCode: 503,
```

Reply:
I had thought that the health check should return 200 to signal that the request completed successfully and the body inspected for a result, otherwise the 503 could mean that the health check could not be reached (which isn't really that important to know). Having considered it again I think returning a 503 on failure does make sense, some health checks may work on status code alone and I have seen that this is how it has been implemented in the other Apollo packages.

Reply:
I can certainly understand that thinking.  The bit about some health check implementations relying on the status code exclusively is the most relevant.  Parsing JSON responses is not functionality that all probes/tools support (or it adds too much configuration to make it worthwhile).

For the above reason, and consistency with the other Apollo implementations (and documentation clarity!) I think we should go with 503 here, unless we think that won't play nice with AWS for some reason.

Reply:
Yes that's fine. Route 53 health checks require the endpoint to respond with an HTTP status code of 2xx or 3xx so that will work.